dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMapper.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import com.datadog.android.core.internal.event.NoOpEventMapper
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.event.EventMapper
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+
+internal data class RumEventMapper(
+    val viewEventMapper: EventMapper<ViewEvent> = NoOpEventMapper(),
+    val errorEventMapper: EventMapper<ErrorEvent> = NoOpEventMapper(),
+    val resourceEventMapper: EventMapper<ResourceEvent> = NoOpEventMapper(),
+    val actionEventMapper: EventMapper<ActionEvent> = NoOpEventMapper()
+) : EventMapper<RumEvent> {
+
+    override fun map(event: RumEvent): RumEvent? {
+        val bundledMappedEvent = when (val bundledEvent = event.event) {
+            is ViewEvent -> viewEventMapper.map(bundledEvent)
+            is ActionEvent -> actionEventMapper.map(bundledEvent)
+            is ErrorEvent -> errorEventMapper.map(bundledEvent)
+            is ResourceEvent -> resourceEventMapper.map(bundledEvent)
+            else -> {
+                sdkLogger.w(
+                    "RumEventMapper: there was no EventMapper assigned for" +
+                        " RUM event type: [${bundledEvent.javaClass.simpleName}]"
+                )
+                bundledEvent
+            }
+        }
+
+        // we need to check if the returned bundled mapped object is not null and same instance as the
+        // original one. Otherwise we will drop the event.
+        return if (bundledMappedEvent != null && bundledMappedEvent === event.event) {
+            event
+        } else {
+            sdkLogger.w(
+                "RumEventMapper: either the returned mapped object was null or was not the " +
```

Comment:
Returning null should not raise a warning (maybe we info) as this is a feature. So maybe you should split this if in two cases